Fatras/include/ActsFatras/Physics/PhotonConversion/PhotonConversion.hpp

```diff
@@ -0,0 +1,350 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/Material/MaterialSlab.hpp"
+#include "ActsFatras/EventData/Particle.hpp"
+#include "ActsFatras/EventData/ProcessType.hpp"
+#include "ActsFatras/Utilities/ParticleData.hpp"
+
+#include <limits>
+#include <random>
+#include <vector>
+
+#include <math.h>
+
+namespace ActsFatras {
+/// @brief This class handles the photon conversion. It evaluates the distance
+/// after which the interaction will occur and the final state due the
+/// interaction itself.
+class PhotonConversion {
+ public:
+  using Scalar = ActsFatras::Particle::Scalar;
+  /// Lower energy threshold for produced children
+  Scalar childMomentumMin = 50. * Acts::UnitConstants::MeV;
+  /// Scaling factor of children energy
+  Scalar childEnergyScaleFactor = 2.;
+  /// Scaling factor for photon conversion probability
+  Scalar conversionProbScaleFactor = 0.98;
+
+  /// @brief Method for evaluating the distance after which the photon
+  /// conversion will occur
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] momentum The momentum of the particle
+  ///
+  /// @param The distance in X_0
+  template <typename generator_t>
+  std::pair<Scalar, Scalar> generatePathLimits(generator_t& generator,
+                                               const Particle& particle) const;
+
+  /// @brief This method evaluates the final state due to the photon conversion
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in, out] particle The interacting photon
+  /// @param [out] generated List of generated particles
+  ///
+  /// @return True if the conversion occured, else false
+  template <typename generator_t>
+  bool run(generator_t& generator, Particle& particle,
+           std::vector<Particle>& generated) const;
+
+ private:
+  /// @brief This method constructs and returns the child particles
+  ///
+  /// @tparam generator_t Type of the random number generator
+  /// @param [in, out] generator The random number generator
+  /// @param [in] photon The interacting photon
+  /// @param [in] childEnergy The energy of one child particle
+  /// @param [in] childDirection The direction of the child particle
+  ///
+  /// @return Vector containing the produced leptons
+  template <typename generator_t>
+  std::vector<Particle> recordProduct(
```

Comment:
```suggestion
  std::vector<Particle> generateChildren(
```
The function does not record anything and creates more than one product.

Reply:
Done